.changeset/slimy-cobras-end.md

```diff
@@ -0,0 +1,7 @@
+---
+"@astrojs/mdx": major
+---
+
+Allows integrations after the MDX integration to update `markdown.remarkPlugins` and `markdown.rehypePlugins`, and have the plugins work in MDX too.
+
+If you rely on the ordering before to not add remark/rehype plugins for MDX, you need to configure `@astrojs/mdx` with `extendMarkdownConfig: false` and explicitly specify the `remarkPlugins` and `rehypePlugins` options instead.
```

Comment:
```suggestion
If you rely on Astro's previous behavior to prevent integrations from adding remark/rehype plugins for MDX, you will now need to configure `@astrojs/mdx` with `extendMarkdownConfig: false` and explicitly specify any `remarkPlugins` and `rehypePlugins` options instead.
```
Would this be accurate?

Reply:
I think I'd tweak it a bit like below, replacing "to prevent" with "that prevents", so it's also targeted towards integration authors that relied on it. I think it's less likely end-users would be affected by this change directly.

```suggestion
If you rely on Astro's previous behavior that prevents integrations from adding remark/rehype plugins for MDX, you will now need to configure `@astrojs/mdx` with `extendMarkdownConfig: false` and explicitly specify any `remarkPlugins` and `rehypePlugins` options instead.
```

Reply:
Ah, great call. If this is targeted at integration authors, I'd like to see that reflected here.

```suggestion
If your integration relies on Astro's previous behavior that prevents integrations from adding remark/rehype plugins for MDX, you will now need to configure `@astrojs/mdx` with `extendMarkdownConfig: false` and explicitly specify any `remarkPlugins` and `rehypePlugins` options instead.
```

What about something like that??

Reply:
Perfect!